src/graph/UpdateEdgeExecutor.cpp

```diff
@@ -0,0 +1,574 @@
+        if (!Expression::isInt(src)) {
+            return Status::Error("SRC Vertex ID should be of type integer");
+        }
+        edge_.set_src(Expression::asInt(src));
+
+        status = sentence_->getSrcId()->prepare();
+        if (!status.ok()) {
+            return status;
+        }
+        auto dst = sentence_->getDstId()->eval();
+        if (!Expression::isInt(dst)) {
+            return Status::Error("DST Vertex ID should be of type integer");
+        }
+        edge_.set_dst(Expression::asInt(dst));
+        edge_.set_ranking(sentence_->getRank());
+
+        auto spaceId = ectx()->rctx()->session()->space();
+        edgeTypeName_ = sentence_->getEdgeType();
+        expCtx_->addAlias(*edgeTypeName_, AliasKind::Edge);
+        yieldExpCtx_->addAlias(*edgeTypeName_, AliasKind::Edge);
+        auto edgeStatus = ectx()->schemaManager()->toEdgeType(spaceId, *edgeTypeName_);
+        if (!edgeStatus.ok()) {
+            status = edgeStatus.status();
+            break;
+        }
+        edgeType_ = edgeStatus.value();
+        schema_ = ectx()->schemaManager()->getEdgeSchema(spaceId, edgeType_);
+        if (schema_ == nullptr) {
+            status = Status::Error("No schema found for '%s'", edgeTypeName_->c_str());
+            break;
+        }
+        edge_.set_edge_type(edgeType_);
+
+        status = prepareSet();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareWhere();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareYield();
+        if (!status.ok()) {
+            break;
+        }
+        status = prepareNeededProps();
+        if (!status.ok()) {
+            break;
+        }
+    } while (false);
+
+    return status;
+}
+
+
+Status UpdateEdgeExecutor::prepareSet() {
+    auto status = Status::OK();
+    auto updateList = sentence_->updateList();
+    if (updateList != nullptr) {
+        auto items = updateList->items();
+        for (auto& item : items) {
+            // item: name_label ASSIGN expression
+            storage::cpp2::UpdateItem updateItem;
+            updateItem.name = *edgeTypeName_;
+            auto propName = item->field();
+            if (schema_->field(*propName) == nullptr) {
+                status = Status::Error("The schema '%s' does not contain the property '%s'",
+                                       edgeTypeName_->c_str(), propName->c_str());
+                break;
+            }
+            updateItem.field = *propName;
+            item->value()->setContext(expCtx_.get());
+            status = item->value()->prepare();
+            if (!status.ok()) {
+                status = Status::Error("Invalid update item value expression: "
+                                       + item->toString());
+                break;
+            }
+            updateItem.value = Expression::encode(item->value());
+            updateItems_.emplace_back(std::move(updateItem));
+        }
+    }
+    return status;
```

Comment:
if updateList == nullptr, maybe should return some error?

Reply:
The updateList always isn't nullptr. 
`auto items = sentence_->updateList()->items();`